### PR TITLE
bpo-1635741: testbuffer heap types

### PR DESCRIPTION
**Pull request by koubaa**
https://bugs.python.org/issue1635741

### CODE REVIEW

**Comment by koubaa** [comment]
@vstinner @shihai1991 @corona10  I tried to take the conversions of just the heap types from #22003 in this PR.  The tests are crashing here but I'm putting it up for review in case any of you could help me spot the problem.

multi-phase init, moving more globals to the module state, and any PEP7 style changes are going to come in a follow-up once I can identify the problems with this change.

File: Modules/_testbuffer.c [hunk 1]
```diff
@@ -2734,7 +2721,9 @@ staticarray_init(PyObject *self, PyObject *args, PyObject *kwds)
 static void
 staticarray_dealloc(StaticArrayObject *self)
 {
+    PyTypeObject *tp = Py_TYPE(self);
     PyObject_Del(self);
```
Comment (shihai1991) [inline Modules/_testbuffer.c:2725]: tp->tp_free(self) in here?

**Review by vstinner — COMMENTED** [review]
See compiler warnings at https://github.com/python/cpython/pull/22131/files , like:



‘ndarray_as_buffer’ defined but not used [-Wunused-variable]

**Comment by skrah** [comment]
This is a test module. The PR removes a test that explicitly *tests static types* at the request of @scoder .

It is not necessary to make changes to this module.

**Pull request closed without merging**